src/main/java/com/microsoft/sqlserver/jdbc/Util.java

```diff
@@ -970,7 +970,7 @@ static synchronized boolean checkIfNeedNewAccessToken(SQLServerConnection connec
 
     // if driver is for JDBC 42 and jvm version is 8 or higher, then always return as SQLServerPreparedStatement42,
     // otherwise return SQLServerPreparedStatement
-    static boolean use42Wrapper() {
+    public static boolean use42Wrapper() {
```

Comment:
Checking for java 7 only doesn't solve the issue - the tests being done on github uses java 8, but does not support jdbc 4.2 - we need to check for both.